test/cctest/test_environment.cc

```diff
@@ -42,6 +42,7 @@ class EnvironmentTest : public NodeTestFixture {
 
     ~Env() {
       FreeIsolateData(isolate_data_);
+      environment_->CleanupHandles();
```

Comment:
I would suggest landing this in a separate commit, it’s a logically not part of the rest of the changes, and we probably want to backport it (e.g. as part of https://github.com/nodejs/node/issues/12664)